keepassxc-browser/common/global.js

```diff
@@ -56,86 +56,54 @@ const ManualFill = {
     BOTH: 2
 };
 
-/**
- * Transforms a valid match pattern into a regular expression
- * which matches all URLs included by that pattern.
- *
- * @param  {string}  pattern  The pattern to transform.
- * @return {RegExp}           The pattern's equivalent as a RegExp.
- * @throws {TypeError}        If the pattern is not a valid MatchPattern
- *
- * https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Match_patterns
- */
-const matchPatternToRegExp = function(pattern) {
-    if (pattern === '') {
-        return (/^(?:http|https|file|ftp|app):\/\//);
+// Match hostname or path with wildcards
+const matchWithRegex = function(firstUrlPart, secondUrlPart, separator) {
+    if (firstUrlPart === secondUrlPart) {
+        return true;
     }
 
-    // special handling of file:// since there is no host
-    if (pattern.startsWith('file://')) {
-        let regex = '^';
-        pattern = pattern.replace(/\./g, '\\.');
-        if (pattern.endsWith('*')) {
-            regex += pattern.slice(0, -1);
-        } else {
-            regex += `${pattern}$`;
-        }
-        return new RegExp(regex);
-    }
-
-    const matchPatternRegExp = new RegExp(
-        `^${schemeSegment}://${hostSegment}/${pathSegment}$`
-    );
-
-    const match = matchPatternRegExp.exec(pattern);
-    if (!match) {
-        throw new TypeError(`"${pattern}" is not a valid MatchPattern`);
+    // If there's no wildcard with hostname, just compare directly
+    if (separator === '.' && !firstUrlPart.includes(URL_WILDCARD) && firstUrlPart !== secondUrlPart) {
```

Comment:
I am not a fan of the separator usage, it is really just a selector for different code paths, it would be better to make it a boolean value or split this function into two, one for hostnames one for paths.